node-upstream-tests/run_tests.js

```diff
@@ -0,0 +1,387 @@
+'use strict'
+
+/* eslint-disable no-console */
+
+const promisify = require('util').promisify
+const axios = require('axios')
+const childProcess = require('child_process')
+const fs = require('fs')
+const glob = promisify(require('glob'))
+const path = require('path')
+
+const NODE_BIN = process.env['NODE_BIN'] || '/usr/bin/node'
+const NODE_REPO_PATH = process.env['NODE_PROJECT']
+if (NODE_REPO_PATH === undefined) {
+  throw new Error('The env variable NODE_PROJECT is not set. This is required to locate the root of the nodejs repo')
```

Comment:
Why not pull the repo directly in this script as well? This would also allow running the tests locally without additional commands that are otherwise only found in the CircleCI configuration.